gomanifest/build.go

```diff
@@ -0,0 +1,234 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+/* MANIFEST FILE VERSION */
+const manifestVersion = "v1"
+
+/* Structure required for parsing `go list -deps -json ./...` json parsing */
+
+// GoPackages ... Package list structure from deps json
+type GoPackages struct {
+	Packages []GoPackage `json:"Packages"`
+}
+
+// GoModule ... Module structure from deps json
+type GoModule struct {
+	Path    string    `json:"Path"`
+	Main    bool      `json:"Main"`
+	Version string    `json:"Version"`
+	Replace *GoModule `json:"Replace"`
+}
+
+// GoPackage ... Package structure from deps json
+type GoPackage struct {
+	Root       string   `json:"Root"`
+	ImportPath string   `json:"ImportPath"`
+	Module     GoModule `json:"Module"`
+	Standard   bool     `json:"Standard"`
+	Imports    []string `json:"Imports"`
+	Deps       []string `json:"Deps"`
+}
+
+/* Structure for output the manifest */
+
+// ManifestTransitiveDeps ... Transitive details in final manifest
+type ManifestTransitiveDeps struct {
+	Name    string `json:"package"`
+	Version string `json:"version"`
+}
+
+// MainfestDirectDeps ... Direct dependency details.
+type MainfestDirectDeps struct {
+	Name         string                   `json:"package"`
+	Version      string                   `json:"version"`
+	Dependencies []ManifestTransitiveDeps `json:"deps"`
+}
```

Comment:
It can be simplified using self referencing struct pointer.
https://play.golang.org/p/Ym8fbkrluBF
```suggestion
	type Dependency struct {
		Name    string        `json:"package"`
		Version string        `json:"version"`
		Deps    []*Dependency `json:"deps"`
	}
```

Reply:
This can be done, but it will generate output with empty 'deps' for transitives, which we don't want.

Reply:
As we discussed, the recursion with omit empty flag will not generate 'deps' attribute for direct dependency in manifest. This use case handle gracefully in API server, so made change and reduced one struct.